app/Console/Commands/FetchRecreatex.php

```diff
@@ -130,6 +137,130 @@ private function handleService()
+                    ->setTime(0, 0, 0)->format('Y-m-d\TH:i:s');
+                $dailyEventList[$date][] = [
+                  'Date' => $date,
+                  'From1' => $reservation['StartDateTime'],
+                  'To1' => $reservation['EndDateTime'],
+                  'From2' => null,
+                  'To2' => null,
+                  'start' => Carbon::createFromFormat('Y-m-d\TH:i:s', $reservation['StartDateTime'])->getTimestamp(),
+                  'end' => Carbon::createFromFormat('Y-m-d\TH:i:s', $reservation['EndDateTime'])->getTimestamp(),
+                ];
+            }
+            $dailyEventList = array_map([$this, 'processCollapsedReservations'], $dailyEventList);
+            $eventList = [];
+            foreach ($dailyEventList as $dayEvents) {
+                $eventList = array_merge($eventList, $dayEvents);
+            }
+            $channel = $this->getOrCreateChannel($channelName);
+            $openinghours = $this->getOrCreateOpeninghours($channel, $year);
+            $calendar = $this->getOrCreateCalendar($openinghours);
+
+            $success = $success && $this->fillCalendar($calendar, $year, $eventList);
+            $processedChannels[] = $channelName;
+        }
+        $channels = Channel::where('service_id', $this->activeServiceRecord->id)
+            ->whereNotIn('label', $processedChannels)
+            ->get();
+        foreach ($channels as $channelToCheck) {
+            $this->clearChannelOpeningHour($channelToCheck, $year);
+        }
+        return $success;
+    }
+
+    protected function getReservations($uuid, $year)
+    {
+        $parameters = [
+            'Context' => [
+                'ShopId' => $this->shopId,
+            ],
+            'ReservationSearchCriteria' => [
+                'InfrastructureId' => $this->activeServiceRecord->identifier,
+                'FromDateTime' => $year . '-01-01T00:00:00.8115784+02:00',
+                'ToDateTime' => $year+1 . '-01-01T00:00:00.8115784+02:00',
+                'ReservationActivityId' => $uuid,
+                'Includes' => [
+                    'SingleReservations' => true,
+                    'ReservationsInList' => true,
+                    'SerieReservations' => true,
+                    'PlaceInfo' => true,
+                    'InfrastructureInfo' => true,
+                    'ReservationActivityInfo' => true,
+                    'ReservedPlaces' => true,
+                ],
+                'Paging' => [
+                    'PageSize' => 999999,
+                    'PageIndex' => 0,
+                ]
+            ],
+        ];
+
+        try {
+            $response = $this->getClient()->FindReservations($parameters);
+            $transformedData = json_decode(json_encode($response), true);
+        } catch (\Exception $e) {
+            $this->error('A problem in collecting external data from Recreatex for ' . $this->activeServiceRecord . ' with year ' .
```

Comment:
Will `$this->activeServiceRecord` be correctly cast to a string?  Could not see a __toString() implementation of any sorts...

Reply:
... that's debatable: https://github.com/illuminate/database/blob/master/Eloquent/Model.php#L1638